llvm/lib/Transforms/InstCombine/InstCombineLoadStoreAlloca.cpp

```diff
@@ -1031,6 +1031,16 @@ Instruction *InstCombinerImpl::visitLoadInst(LoadInst &LI) {
         return replaceOperand(LI, 0, SI->getOperand(1));
     }
   }
+
+  if (Triple(LI.getModule()->getTargetTriple()).isXilinxFPGA()) {
```

Comment:
Add rationale as comments

Reply:
And looking again at it, would it not be clearer to have this into onr of our vinification (Vitis...) passes instead of scattered across some random passes, when possible?